scripts/build/autotools/CheckSSL.m4

```diff
@@ -0,0 +1,240 @@
+
+    if test "$have_ssl_headers" = "yes"; then
+      PHP_EVAL_INCLINE([$PHP_MONGODB_SSL_CFLAGS])
+      PHP_EVAL_LIBLINE([$PHP_MONGODB_SSL_LIBS],[MONGODB_SHARED_LIBADD])
+      PHP_MONGODB_SSL="openssl"
+      found_openssl="yes"
+    elif test "$PHP_MONGODB_SSL" = "openssl"; then
+      AC_MSG_ERROR([OpenSSL development headers could not be found])
+    fi
+
+    AC_CHECK_DECLS([ASN1_STRING_get0_data],
+                   [have_ASN1_STRING_get0_data="yes"],
+                   [have_ASN1_STRING_get0_data="no"],
+                   [[#include <openssl/asn1.h>]])
+
+    CFLAGS="$old_CFLAGS"
+  ])
+
+  if test "$found_openssl" = "no"; then
+    unset OPENSSL_INCDIR
+    unset OPENSSL_LIBDIR
+
+    if test "$PHP_OPENSSL_DIR" = "yes"; then
+      PHP_OPENSSL_DIR="/usr/local/ssl /usr/local /usr /usr/local/openssl"
+    fi
+
+    for i in $PHP_OPENSSL_DIR; do
+      if test -r $i/include/openssl/evp.h; then
+        OPENSSL_INCDIR="$i/include"
+      fi
+      if test -r $i/$PHP_LIBDIR/libssl.a -o -r $i/$PHP_LIBDIR/libssl.$SHLIB_SUFFIX_NAME; then
+        OPENSSL_LIBDIR="$i/$PHP_LIBDIR"
+      fi
+      test -n "$OPENSSL_INCDIR" && test -n "$OPENSSL_LIBDIR" && break
+    done
+
+    if test -n "$OPENSSL_LIBDIR"; then
+      OPENSSL_LIBDIR_LDFLAG="-L$OPENSSL_LIBDIR"
+    fi
+
+    PHP_CHECK_LIBRARY([crypto],
+                      [EVP_DigestInit_ex],
+                      [have_crypto_lib="yes"],
+                      [have_crypto_lib="no"],
+                      [$OPENSSL_LIBDIR_LDFLAG])
+    PHP_CHECK_LIBRARY([ssl],
+                      [SSL_library_init],
+                      [have_ssl_lib="yes"],
+                      [have_ssl_lib="no"],
+                      [$OPENSSL_LIBDIR_LDFLAG -lcrypto])
+
+    if test "$have_ssl_lib" = "yes" -a "$have_crypto_lib" = "yes"; then
+      PHP_ADD_LIBRARY([ssl],,[MONGODB_SHARED_LIBADD])
+      PHP_ADD_LIBRARY([crypto],,[MONGODB_SHARED_LIBADD])
+
+      if test -n "$OPENSSL_LIBDIR"; then
+        PHP_ADD_LIBPATH([$OPENSSL_LIBDIR],[MONGODB_SHARED_LIBADD])
+      fi
+
+      if test -n "$OPENSSL_INCDIR"; then
+        PHP_ADD_INCLUDE($OPENSSL_INCDIR)
+      fi
+
+      old_CFLAGS="$CFLAGS"
+      CFLAGS="$OPENSSL_INCDIR $CFLAGS"
+
+      AC_CHECK_DECLS([ASN1_STRING_get0_data],
+                     [have_ASN1_STRING_get0_data="yes"],
+                     [have_ASN1_STRING_get0_data="no"],
+                     [[#include <openssl/asn1.h>]])
+
+      CFLAGS="$old_CFLAGS"
+
+      PHP_MONGODB_SSL="openssl"
+      found_openssl="yes"
+    fi
+  fi
+
+  if test "$PHP_MONGODB_SSL" = "openssl" -a "$found_openssl" != "yes"; then
+    AC_MSG_ERROR([OpenSSL libraries and development headers could not be found])
+  fi
+])
+
+AS_IF([test "$PHP_MONGODB_SSL" = "libressl" -o "$PHP_MONGODB_SSL" = "auto"],[
+  found_libressl="no"
+
+  dnl TODO: this may require adding libcrypto to the modules list, as pkg-config does not seem to include it (unlike openssl)
+  PKG_CHECK_MODULES([PHP_MONGODB_SSL],[libtls],[
```

Comment:
Note: when testing this on my Ubuntu system, I actually had to change "libtls" to "libressl-tls" (since I'm using [a custom PPA](https://launchpad.net/~strukturag/+archive/ubuntu/libressl)). I also noticed that `pkg-config --libs libressl-tls` returns only `-lressl-tls`, unlike `pkg-config --libs openssl`, which returns `-lssl -lcrypto`. I'm not sure if that's specific to the PPA package I'm using, so I'll have to test with LibreSSL proper.

FWIW, libmongoc also only specifics "libtls", so I'm curious if their LibreSSL builds are linking `-lcrypto` or not.

Reply:
It's possible that your PPA doesn't need the -lcrypto, and just that -lressl-tls contains all of what -lssl and -lcrypt export. I would actually go as far to make sure that we can support your custom PPA, and the normal libressl.